cpp/dataStructures/CMakeLists.txt

```diff
@@ -1,16 +1,12 @@
-cmake_minimum_required(VERSION 3.2.2)
-set(CMAKE_BUILD_TYPE Release)
-
-#For the shared library:
-set ( PROJECT_LINK_LIBS libcppCommon.dylib )
-link_directories( ../common/build )
+# However, the file(GLOB...) allows for wildcard additions:
+file(GLOB SOURCES "src/*.cpp")
 
-#Bring the headers, such as Student.h into the project
-include_directories(include)
-include_directories(../common/include/)
+# Generate the shared library from the sources
+add_library(cppDS SHARED ${SOURCES})
 
```

Comment:
Funnily enough, that _is_ something I've already [answered on StackOverflow](http://stackoverflow.com/a/10672739/2556117).

To be honest, I always try and use static libraries rather than shared.  If you change your `add_library` calls to not specify `SHARED`, this would be a non-issue.